### DISCUSSION TITLE
Several versions of poetry at the same time?

### DISCUSSION THREAD

**Discussion by MaximMukhametov** [python-poetry/poetry#7866] [Q&A] [answered]
I work in parallel on several projects, all projects use poetry, but everywhere there is a different version, how can I keep several versions of poetry locally at the same time?

**Comment by finswimmer** [comment, marked as answer]
I'd recommend using [pipx](https://pypa.github.io/pipx/) to install and use multiple versions of Poetry in parallel.

```
$ pipx install --suffix "@1.4" poetry==1.4.2
$ pipx install --suffix "@1.3" poetry==1.3.5

$ poetry@1.4 --version
Poetry (version 1.4.2)
$ poetry@1.3 --version
Poetry (version 1.3.5)
```

But I wonder why you need several versions?

**Reply by MaximMukhametov** [reply]
> But I wonder why you need several versions?

As I said, I work in parallel on several projects, so my lock files are incompatible with poetry version in containers on which applications are deployed

**Comment by petroslamb** [comment]
Hi there, just wondering, did this work well with the poetry caches? Do the two installations manage to collaborate on the cache successfully?

**Comment by cdddg** [comment]
I’ve seen a few related discussions — [#4235](https://github.com/python-poetry/poetry/issues/4235), [#8811](https://github.com/python-poetry/poetry/issues/8811), and now this one #7866.

While it’s not a core feature request, managing multiple Poetry versions becomes quite painful once you actually need to do so.

I ran into the same issue at work, so I built a small tool called [poetryenv](https://github.com/cdddg/poetryenv)
 — its usage and style are inspired by pyenv.

It lets you install, switch, and pin specific Poetry versions per project.
Hope it helps others with the same need!

**Reply by brienpacholec** [reply]
This is awesome!
